src/tests/ftest/util/daos_io_conf.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2019 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+import os
+
+from apricot import TestWithServers
+from command_utils import ExecutableCommand
+from test_utils import TestPool
+
+class IoConfGen(ExecutableCommand):
```

Comment:
If daos_io_conf is a bin command with options and as you are already inheriting ExecutableCommand class, if you don't mind lets take full advantage of it. I think that whole code can be simplified and executed a bit differently. You can have a look at dfuse_utils.py/mdtest_utils.py for example.

Most of your gen_io_conf() params can go inside init() method of your IoConfGen() class. I am not familiar with the daos_io_conf bin, but if those are the options of that bin then it makes sense to have it in the init()

Reply:
Well the init() will be more useful if we intended to use single test case with Avocado !mux, which is the case right now but looking for the future we need to use the same utility to the extend where we need to generate multiple files for single tests so that was the reason I created IoConfTestBase for run and create.